ui/src/app/pages/data/concept/concept-add-modal.tsx

```diff
@@ -96,13 +109,39 @@ export const ConceptAddModal = withCurrentWorkspace()
     }
   }
 
+  isSurvey() {
+    return this.props.selectedDomain.domain === Domain.OBSERVATION;
+  }
+
   async saveConcepts() {
-    const {workspace: {namespace, id}} = this.props;
+    const {workspace: {namespace, id}, selectedSurvey, surveyName} = this.props;
     const {onSave, selectedDomain} = this.props;
-    const {selectedSet, addingToExistingSet, newSetDescription,
-      name, selectedConceptsInDomain} = this.state;
+    const {
+      selectedSet, addingToExistingSet, newSetDescription,
+      name, selectedConceptsInDomain
+    } = this.state;
     this.setState({saving: true});
-    const conceptIds = fp.map(selected => selected.conceptId, selectedConceptsInDomain);
+    let conceptIds = fp.map(selected => selected.conceptId, selectedConceptsInDomain);
+    let survey = Surveys.THEBASICS;
+    if (this.isSurvey()) {
+      conceptIds = selectedSurvey.map((surveys) => surveys.conceptId);
+
+      switch (surveyName) {
```

Comment:
This seems like its pretty likely to be reused elsewhere. (And maybe is?) 

Can we either make this a map or a common util? (Probably utils for enum->displayname, and vice versa)